### PR TITLE
Resolve documentation problems identified in issues #2136, #872, #993, and #2194

### PR DESCRIPTION
**Pull request by olyson**
### Description of changes

Resolve documentation problems identified in issues #2136, #872, #993, #2194

Contributors other than yourself, if any:  @samsrabin 

CTSM Issues Fixed (include github issue #): #2136 , #872, #993, #2194

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any:
Built html on local machine.

### CODE REVIEW

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst [hunk 1]
```diff
@@ -533,39 +533,41 @@ The soil moisture deficit :math:`D_{irrig}` is
 
    D_{irrig} = \left\{
    \begin{array}{lr}
-   w_{thresh} - w_{avail} &\qquad w_{thresh} > w_{avail} \\
+   w_{target} - w_{avail} &\qquad w_{thresh} > w_{avail} \\
    0 &\qquad w_{thresh} \le w_{avail}
    \end{array} \right\}
 
-where :math:`w_{thresh}` is the irrigation moisture threshold (mm) and :math:`w_{avail}` is the available moisture (mm). The moisture threshold is
+where :math:`w_{target}` is the irrigation target soil moisture (mm)
```
Comment (samsrabin) [inline doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:540]: - [ ] Could you add back the definition of $w_{avail}$ here? It doesn't seem to be defined anywhere else.

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst [hunk 2]
```diff
@@ -533,39 +533,41 @@ The soil moisture deficit :math:`D_{irrig}` is
...
+where :math:`w_{wilt}` is the wilting point soil moisture (mm)
 
 .. math::
    :label: 25.64
 
    w_{wilt} = \sum_{j=1}^{N_{irr}} \theta_{wilt} \Delta z_{j} \ ,
 
 and :math:`f_{thresh}` is a tuning parameter.  The available moisture in
-the soil is
+the soil (mm) is
```
Comment (samsrabin) [inline doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:562]: - [ ] Looks like I missed the manual line break here. Could you please move this line up to the end of the previous one?

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst [hunk 3]
```diff
@@ -533,39 +533,41 @@ The soil moisture deficit :math:`D_{irrig}` is
...
 
 .. math::
    :label: 25.65
 
    w_{avail} = \sum_{j=1}^{N_{irr}} \theta_{j} \Delta z_{j} \ ,
 
+Note that :math:`w_{target}` is truly supposed to give the target soil moisture value that we're shooting for whenever irrigation happens; then the soil moisture deficit :math:`D_{irrig}` gives the difference between this target value and the current soil moisture. The irrigation moisture threshold :math:`w_{thresh}`, on the other hand, gives a threshold at which we decide to do any irrigation at all. The way this is written allows for the possibility that one may not want to irrigate every time there becomes even a tiny soil moisture deficit. Instead, one may want to wait until the deficit is larger before initiating irrigation; at that point, one doesn't want to just irrigate up to the "threshold" but instead up to the higher "target". The target should always be greater than or equal to the threshold.
```
Comment (samsrabin) [inline doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:569]: Great explanation!

File: doc/source/tech_note/Land-Only_Mode/CLM50_Tech_Note_Land-Only_Mode.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.. _rst_Land-only Mode:
+ _rst_Land-only Mode:
```
Comment (samsrabin) [inline doc/source/tech_note/Land-Only_Mode/CLM50_Tech_Note_Land-Only_Mode.rst:1]: - [ ] This change results in `undefined label: rst_land-only mode`. Could you please undo? (Or was there a reason you did this?)

File: doc/source/tech_note/Photosynthetic_Capacity/CLM50_Tech_Note_Photosynthetic_Capacity.rst [hunk 5]
```diff
@@ -53,65 +53,65 @@ The structural nitrogen, :math:`N_{\text{str}}`, is calculated as the multiplica
...
+ | Tropical Corn                    |          0.05000         |
+ +----------------------------------+--------------------------+
+ | Tropical Soybean                 |          0.03500         |
+ +----------------------------------+--------------------------+
+ | Miscanthus                       |          0.03500         |
+ +----------------------------------+--------------------------+
+ | Switchgrass                      |          0.03500         |
+ +----------------------------------+--------------------------+
+
+Notes: :math:`SLA_{\text{0}}` is the specific leaf area at the canopy top (m :sup:`2` leaf/g biomass)
```
Comment (samsrabin) [inline doc/source/tech_note/Photosynthetic_Capacity/CLM50_Tech_Note_Photosynthetic_Capacity.rst:114]: - [ ] I actually refer to $N_{cb}$ in #2221. Is there a reason you got rid of it here?

**Review by samsrabin — CHANGES_REQUESTED** [review]
Thanks for all these changes, Keith! It'll be great to get these four issues resolved. I just have some minor requests before approval.

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst (hunk 1 above)
Comment (olyson) [reply doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:540]: It is defined just below in equation 2.26.23, least in my html build.

File: doc/source/tech_note/Land-Only_Mode/CLM50_Tech_Note_Land-Only_Mode.rst (hunk 4 above)
Comment (olyson) [reply doc/source/tech_note/Land-Only_Mode/CLM50_Tech_Note_Land-Only_Mode.rst:1]: Good catch.  Not sure how that happened.  Fixed now (but please double-check).

File: doc/source/tech_note/Photosynthetic_Capacity/CLM50_Tech_Note_Photosynthetic_Capacity.rst (hunk 5 above)
Comment (olyson) [reply doc/source/tech_note/Photosynthetic_Capacity/CLM50_Tech_Note_Photosynthetic_Capacity.rst:114]: It's not actually a parameter in the parameter file.

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst (hunk 3 above)
Comment (olyson) [reply doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:569]: Actually, that was @billsacks 's great explanation.

File: doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst (hunk 2 above)
Comment (olyson) [reply doc/source/tech_note/Crop_Irrigation/CLM50_Tech_Note_Crop_Irrigation.rst:562]: Looks like I caused this.  Fixed now.

**Review by samsrabin — APPROVED** [review]

**Comment by samsrabin** [comment]
Thanks, Keith!

**Pull request merged**